### DISCUSSION TITLE
How to add different styles of multiline text

### DISCUSSION THREAD

**Discussion by a123likeyou** [ImageMagick/ImageMagick6#118] [Q&A]
![333](https://user-images.githubusercontent.com/22471187/101580165-4cac0c00-3a15-11eb-8e49-fa1276c4e324.png)
I want to implement a command to add multiple lines of text, and it is possible that the style of each text is different (font color and some other styles). How can I do it most conveniently and quickly? I have thought for a long time

**Comment by urban-warrior** [comment]
You want the Pango Markup Language.  See https://legacy.imagemagick.org/Usage/text/#pango.

**Reply by a123likeyou** [reply]
Thank you for your answer, but we need to customize the upload font. It seems that pango cannot use custom fonts, and we are using java language

**Reply by a123likeyou** [reply]
> You want the Pango Markup Language. See https://legacy.imagemagick.org/Usage/text/#pango.

Thank you for your answer, but we need to customize the upload font. It seems that pango cannot use custom fonts, and we are using java language

**Reply by flawrist** [reply]
@urban-warrior, Pango allows you to use different fonts and text styles in the same command indeed, but it has a problem with vertical alignment since `-gravity center` in its case centers only horizontally. IM 7.1.1.-21.
